lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/TaskStatusAlarmHandler.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runime.azbatch.driver;
+
+import com.microsoft.azure.batch.protocol.models.CloudTask;
+import org.apache.reef.runime.azbatch.util.AzureBatchHelper;
+import org.apache.reef.runime.azbatch.util.TaskStatusMapper;
+import org.apache.reef.runtime.common.driver.resourcemanager.ResourceStatusEvent;
+import org.apache.reef.runtime.common.driver.resourcemanager.ResourceStatusEventImpl;
+import org.apache.reef.tang.InjectionFuture;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.Alarm;
+import org.apache.reef.runtime.common.driver.evaluator.pojos.State;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/*
+ * Class that gets that status of the tasks from Azure Batch for the job that is currently in progress
+ * and notifies REEF of the status.
+ */
+final class TaskStatusAlarmHandler implements EventHandler<Alarm>, AutoCloseable {
```

Comment:
Let's rename this to AzureBatchTaskStatusAlarmHandler so it's not confused with REEF tasks.

Reply:
Done.